samza-sql/src/main/java/org/apache/samza/sql/interfaces/SourceResolver.java

```diff
@@ -41,4 +46,12 @@
    *  true if the source is a table, else false.
    */
   boolean isTable(String sourceName);
+
+  /**
+   * Return a Samza table descriptor corresponding to the {@code sourceName} if the underlying source is a table,
+   * ie. {@link SourceResolver#isTable(String)} should return true for the same {@code sourceName}.
+   * @param sourceName source for which the table descriptor is requested.
+   * @return table descriptor; null if the source is not a table.
+   */
+  TableDescriptor getTableDescriptor(String sourceName);
```

Comment:
Can you merge getTableDescriptor and isTable with SqlSystemSourceConfig. Probably we could call SqlSystemSourceConfig as SamzaSqlIOConfig?

Reply:
>To do this, we would have to move the local table code in join translator into all source resolvers. 
@pdu-mn1 @atoomula That makes sense. Can we do quick offline review to see how this code looks? If it is becoming too big of a change. I am ok with merging this PR and doing a fresh PR for that refactor.